### DISCUSSION TITLE
can't override create hook in api/xxx/controllers

### DISCUSSION THREAD

**Discussion by brtka** [strapi/strapi#7174] [Legacy Topics]
Hi all,

I made a clean install of strapi v3.1.1 after the same issue started apearing in my upgraded version. I was following tutorial on strapi website [Create is owner policy](https://strapi.io/documentation/v3.x/guides/is-owner.html#apply-the-author-by-default), and couldn't trigger the async create hook

**Steps to reproduce the behavior**

1. fresh install of strapi with mongodb with `yarn create strapi-app web_node_12`
2. do not install any other plugins
3. create a new collection named `article` with fields `title - text`,  `author` with relation `User has many Articles` and save
4. paste the exact code to `/api/article/controllers/article.js`

**Code snippets**

```
const { parseMultipartData, sanitizeEntity } = require('strapi-utils');

module.exports = {
  /**
   * Create a record.
   *
   * @return {Object}
   */

  async create(ctx) {
    console.log('create hook called', ctx)
    let entity;
    if (ctx.is('multipart')) {
      const { data, files } = parseMultipartData(ctx);
      data.author = ctx.state.user.id;
      entity = await strapi.services.article.create(data, { files });
    } else {
      ctx.request.body.author = ctx.state.user.id;
      entity = await strapi.services.article.create(ctx.request.body);
    }
    return sanitizeEntity(entity, { model: strapi.models.article });
  },
};
```

5. create an article with a title

**Expected behavior**
If everything is ok, I should set the current logged in user as author. Also in the console I should see printed ctx.
Author is not set, console is empty

**Screenshots**
Articles Collection:
![image](https://user-images.githubusercontent.com/7488012/88464995-8560a400-cebf-11ea-8cf9-03c397d28faa.png)

Applied changes to `/api/article/controllers/article.js`
![image](https://user-images.githubusercontent.com/7488012/88465068-2a7b7c80-cec0-11ea-93ef-d57ceba6f0ad.png)

Created article
![image](https://user-images.githubusercontent.com/7488012/88465021-b50fac00-cebf-11ea-8bf3-5c2a928c127b.png)

Missing console output, or setting of author
![image](https://user-images.githubusercontent.com/7488012/88465095-60206580-cec0-11ea-836f-4a9e88de9ab9.png)

**System**

- Node.js version: v12 LTS
- NPM version: yarn 1.22
- Strapi version: 3.11
- Database: Mongo
- Operating system: MacOS 10.15.5 (19F101)

**Additional Notes**

Of course I've tried just returning in the create method, but it still created an item. I've also tried changing the syntax of the method to 

```
... {
 create: async ctx ()=>{
   console.log(ctx)
   return
}
}

```
Also, I've tried to intercept the service create method where I also just returned and printed data param, the result was same. Than I made a new fresh install with --quick-start option hoping that mongodb is the problem, but still no luck...

Am I missing something, is the provided documentation in tutorial obsolete, or is this a bug?

Thx

**Comment by lauriejim** [comment]
Moved this as a discussion.

It's normal, you updated the controller function, and the controller function is called when you trigger the generated API `POST /articles` in your case.
Here you created your article with the admin panel. And the Content Manage has its own functions to manage the content.

Generated API !== admin panel

I hop it helps you to understand the thing.

**Reply by derrickmehaffy** [reply]
To add a little context, the controllers used by the admin are located here: https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-content-manager/controllers/ContentManager.js

You can customize these using the [extensions system](https://strapi.io/documentation/v3.x/getting-started/troubleshooting.html#how-do-i-customize-a-plugin)

**Reply by brtka** [reply]
I guess I didn't realize that admin POST goes through a different route. Maybe I didn't read the documentation carefully enough. But, it makes sense, that you need to modify content manager directly to change it's behavior. 

Thank you very much for clarification.
